lib/XML/LibXML/Attr.pm

```diff
@@ -25,14 +31,46 @@ method value() {
 }
 
 method type() {
+    nqp::nativecallrefresh(self);
     xmlElementType(nqp::p6box_i(nqp::getattr_i(nqp::decont(self), xmlAttr, '$!type')))
 }
 
 method firstChild() {
+    nqp::nativecallrefresh(self);
     # We cannot use XML::LibXML::Node here directly because Node.pm 'use's Attr.pm.
     nativecast(::('XML::LibXML::Node'), self.children)
 }
 
 multi method gist(XML::LibXML::Attr:D:) {
     self.name ~ '="' ~ self.value ~ '"'
 }
+
+method toString(XML::LibXML::Attr:D:) {
+    self.gist();
+}
+
+# cw: See the need to break out similar operations for all XML nodes into a specific role 
+#     (ala XML::LibXML::Nodish) otherwise we will end up duplicating functionality across
+#     several classes.
+
+method isSameNode($n) {
+    # cw: Maybe.
+
+    # cw: Not. The issue here is that P6 is creating these objects from 
+    #     the backing C-pointer which is lost in the shuffle. So two objects
+    #     can be created from the -same- pointer but have no sense of 
+    #     equivalence (unless... possibly... the repr is 'CPointer')
+    return self =:= $n;
+}
+
+method ownerDocument {
+    return nativecast(::('XML::LibXML::Document'), self.doc);
```

Comment:
Agreed. I only use them to prevent circular "use" references.

Maybe I can break some of these routines out to other areas where they are not necessary.